### PR TITLE
ConvQuestions benchmark added

### PR DESCRIPTION
**Pull request by PhilippChr**
Hello,
I would like to integrate our dataset on conversational QA. The answers are grounded in the KG.
The work was published in CIKM 2019 (https://dl.acm.org/doi/10.1145/3357384.3358016).
We hope for further research on how to deal with the challenges of factoid conversational QA.
Thanks! :)

### CODE REVIEW

File: datasets/conv_questions/conv_questions.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+fair comparison across diverse methods. The data gathering setup was kept as natural as
+possible, with the annotators selecting entities of their choice from each of the five domains,
+and formulating the entire conversation in one session. All questions in a conversation are
+from the same Turker, who also provided gold answers to the questions. For suitability to knowledge
+graphs, questions were constrained to be objective or factoid in nature, but no other restrictive
+guidelines were set. A notable property of ConvQuestions is that several questions are not
+answerable by Wikidata alone (as of September 2019), but the required facts can, for example,
+be found in the open Web or in Wikipedia. For details, please refer to our CIKM 2019 full paper
+(https://dl.acm.org/citation.cfm?id=3358016).
+"""
```
Comment (lhoestq) [inline datasets/conv_questions/conv_questions.py:31]: could you also add all the above info in the corresponding sections of the dataset card please ? :)

File: datasets/conv_questions/conv_questions.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+_LICENSE = "MIT License"
+
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URL = "http://qa.mpi-inf.mpg.de/convex/"
+_URLs = {
+    "train": _URL + "/ConvQuestions_train.zip",
+    "dev": _URL + "/ConvQuestions_dev.zip",
+    "test": _URL + "/ConvQuestions_test.zip",
+}
```
Comment (lhoestq) [inline datasets/conv_questions/conv_questions.py:68]: there is already a "/" at the end of _URL

```suggestion
_URLs = {
    "train": _URL + "ConvQuestions_train.zip",
    "dev": _URL + "ConvQuestions_dev.zip",
    "test": _URL + "ConvQuestions_test.zip",
}
```

File: datasets/conv_questions/conv_questions.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    # data = datasets.load_dataset('conv_questions', 'dev')
+    # data = datasets.load_dataset('conv_questions', 'test')
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="train", version=VERSION, description="Train set of the benchmark."),
+        datasets.BuilderConfig(name="dev", version=VERSION, description="Validation set of the benchmark."),
+        datasets.BuilderConfig(name="test", version=VERSION, description="Test set of the benchmark."),
+    ]
+
+    DEFAULT_CONFIG_NAME = "train"  # It's not mandatory to have a default configuration. Just use one if it make sense.
+
```
Comment (lhoestq) [inline datasets/conv_questions/conv_questions.py:95]: You can remove all this. We distinguish configurations and splits. Configurations can be used when there are several formats/tasks/processing for you dataset. We use splits for train/test/validation separation. In your case you don't need to have configurations.

To load one split one can simply do
```python
load_dataset("conv_questions", split="train")
```

```suggestion
```

File: datasets/conv_questions/README.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  ]
+}
+```
+
+### Data Fields
+
+- `domain`: a `string` feature.
+- `questions`: a `list` of `string` features.
+- `answers`: a `list` of `lists` of `string` features.
+- `answer_texts`: a `list` of `string` features.
```
Comment (lhoestq) [inline datasets/conv_questions/README.md:119]: can you mention that the answers are not provided in the test set ?

File: datasets/conv_questions/conv_questions.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    def _info(self):
+        # This method specifies the datasets.DatasetInfo object which contains informations and typings for the dataset
+        features = datasets.Features(
+            {
+                "domain": datasets.Value("string"),
+                "questions": datasets.features.Sequence(datasets.Value("string")),
+                "answers": datasets.features.Sequence(datasets.features.Sequence(datasets.Value("string"))),
+                "answer_texts": datasets.features.Sequence(datasets.Value("string")),
+            }
+        )
```
Comment (lhoestq) [inline datasets/conv_questions/conv_questions.py:105]: maybe we can also have the seed_entity and seed_entity_text ?

**Review by lhoestq — COMMENTED** [review]
Nice thank you for adding it :)
Good job with the script and the dataset card

I left a few comments.

Also in one of my comments I suggest to remove the configurations. After that you'll have to re-generate the dataset_infos.json file as well as the dummy data files. You can delete the old dataset_infos.json and the old dummy data files.

**Comment by PhilippChr** [comment]
Thanks for your helpful comments and suggestions! :)
I integrated the additional fields, and extended some of the README/dataset card.
And I actually realized that we had the cc-by-4.0 for the dataset, so this was also changed.

**Review by lhoestq — APPROVED** [review]
It looks all good now, thank you ! :)

I just added a line in the readme to say that the answers are not present in the test set

**Comment by PhilippChr** [comment]
I added the answers to the test set actually :)

**Comment by lhoestq** [comment]
Oh great ! Let me revert my change then

**Pull request merged**